### PR TITLE
4.0.0 week numbers

### PR DESCRIPTION
**Pull request by plumpNation**
Here is a start on the week numbers functionality we talked about. I stopped when I got a failing unit test linked to shadow.js so I could get some feedback from you.

This is related to a [pull request on the older version of the library](https://github.com/amsul/pickadate.js/pull/371), that you requested be ported to 4.0.0

### CODE REVIEW

**Comment by DanielRuf** [comment]
@plumpNation can you rebase this on the current tip of the master branch?

**Comment by plumpNation** [comment]
Maybe a silly question, but this code is built on top of tag 4.0.0. Is master also taking in v4 changes?

We did create a PR https://github.com/amsul/pickadate.js/pull/459 for an older version, which looks like it was merged into dev.

**Comment by amsul** [comment]
v4 is definitely not ready for a merge into master.

@DanielRuf you and I should discuss plans for v4 soon-ish 🙂

**Comment by plumpNation** [comment]
Ok, rebase done against the 4.0.0 branch.

**Comment by DanielRuf** [comment]
I guess there is a part missing in the class definition.

```
ReferenceError: The `showWeekNumbers` property is not recognized by {class Pickadate:Picker:DataElement:Element:Object}. in file:///home/travis/build/amsul/pickadate.js/lib/js/vendor/shadow.js (line 511)
```

Let me know if you need any support with this.

You can also run the tests locally with `npm test` afaik.

**Comment by plumpNation** [comment]
Quick update.

tl;dr I will attempt to mend using the repo travis CI test runner.

I can see from the travis CI output that the build is running on a very old version of node.

Notes.
- The package file does not include all the dependencies required to run the build.
- The gitignore does not ignore bower_components, the .grunt folder or the _SpecRunner.html file.
- `npm test` script will not run without grunt-cli. (That hasn't been updated in 2 years and still suggests a global install even though it is not at all necessary.)

When I install grunt-cli so that the `npm test` script will run, I get through the linting, but the tests don't run, failing with this error.

```
Running PhantomJS...ERROR
>> 0 [ '' ]
```

I tried installing phantomjs globally, but the tests just hung without running or throwing an error.
I tried installing phantomjs@1.9.0-1 globally (that is required by grunt-lib-phantomjs in this project), but npm failed to install it.

High Sierra

nvm installed:
node 0.10.28
npm 1.4.9

**Comment by DanielRuf** [comment]
> I tried installing phantomjs@1.9.0-1 globally (that is required by grunt-lib-phantomjs in this project), but npm failed to install it.

Did you try `phantomjs-prebuilt`?

**Comment by DanielRuf** [comment]
https://github.com/amsul/pickadate.js/pull/554/files#diff-de00f6e71314a85b2525c2ae6ca4a209R1238

Shouldn't be this in `attrs`?

File: tests/spec/ui-pickadate.js [hunk 1]
```diff
@@ -1228,6 +1228,32 @@ describe('shadow.Pickadate', function() {
...
     })
 
+    describe('.createGridHeadDates() with showWeekNumbers', function() {
+
+        var pickadate = shadow.Pickadate.create({
+            $el: $('<div />'),
+            attrs: {
+                view: [2003, 2, 1]
+            },
+            showWeekNumbers: true
```
Comment (DanielRuf) [inline tests/spec/ui-pickadate.js]: Move this into the `attrs` property.

**Comment by plumpNation** [comment]
I am actually trying that now, if you can believe that! Appreciate the hint though :D I will try the prebuilt idea too. Thanks.

**Comment by plumpNation** [comment]
```
npm install -g phantomjs-prebuilt
```

Error: 
```
const Hoek = require('hoek');
^^^^^
SyntaxError: Use of const in strict mode.
```

Which version of node are you running locally?

**Comment by DanielRuf** [comment]
Current master uses 8.11 and has some updated package.json and .tracis-ci.yml afaik.

**Comment by DanielRuf** [comment]
For reference: https://travis-ci.org/amsul/pickadate.js/builds/361301715

**Comment by plumpNation** [comment]
Absolutely, I am using nvm.

There are too many issues, for me, running the jshint and tests. I just switched out to a modern node and did a full clean out of the node_modules and globals required and still had issues.

Yes, wk should definitely be in a different position, but not in the array I am using as an expectation. The `gridHeadDates.textContent` is not as expected. `Wk` should be first.

I am making a small change in the way the headers fragment is created to see if that will produce a different result.

**Comment by plumpNation** [comment]
OK, so it looks like that worked, but so far I haven't figured out why, and without my local build working properly it's hard to debug.

I tested locally in the browser and couldn't figure out what was wrong with doc fragment `insertBefore`.

I was using this:
`gridHeaders.insertBefore( createWeekNumber(), gridHeaders.firstElementChild )`
but it seems that that method failed somehow on the travis CI build but not locally for me in browser.

So I changed where the fragment was made and added the week number to it first using the trusty `appendChild`.
